server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetResolver.java

```diff
@@ -261,21 +280,107 @@ private byte[] getFixedLenByteArray(String value, Type type) {
         DecimalLogicalTypeAnnotation typeAnnotation = (DecimalLogicalTypeAnnotation) type.getLogicalTypeAnnotation();
         int precision = Math.min(HiveDecimal.MAX_PRECISION, typeAnnotation.getPrecision());
         int scale = Math.min(HiveDecimal.MAX_SCALE, typeAnnotation.getScale());
+
+        /*
+        When this column is defined as NUMERIC(precision,scale), GPDB will handle writing values with overflow.
+        When this column is defined as NUMERIC, the column type will be treated as DECIMAL(38,18),
+        and HiveDecimal.create has different behaviors for different types of overflow:
+
+        (1) When the data integer digit count is greater than 38,
+        HiveDecimal.create will return a null value. To make the behavior consistent with Hive's behavior
+        when storing on a Parquet-backed table, we store the value as null.
+        For example, the integer digit count of 1234567890123456789012345678901234567890.123 is 40,
+        which is greater than 38. HiveDecimal.create returns null.
+
+        (2) When data integer digit count is not greater than 38,
+        and the overall data precision is greater than 38,
+        HiveDecimal.create will return a rounded-off value to fit in the Hive maximum supported precision 38.
+        For example, the integer digit count of 1234567890123456789012345.12345678901234567890 is 25 which is less than 38,
+        but its overall precision is 45 which is greater than 38.
+        Then data will be created as a rounded value 1234567890123456789012345.1234567890123
+
+        (3) When data integer digit count is not greater than 38,
+        and the overall data precision is not greater than 38,
+        HiveDecimal.create will return the same decimal value as provided.
+        For example, 123456.123456 can fit in DECIMAL(38,18) without and data loss, so the data will be created as the same
+         */
+        // HiveDecimal.create will return a decimal value which can fit in DECIMAL(38)
+        HiveDecimal parsedValue = HiveDecimal.create(value);
+
+        if (parsedValue == null) {
+            if (isDecimalOverflowOptionError || isDecimalOverflowOptionRound) {
+                throw new UnsupportedTypeException(String.format("The integer digit count of data %s in " +
+                                "a NUMERIC column exceeds the maximum supported precision %d. Query failed.",
```

Comment:
we need to report columnName, declare as function parameter, if needed.
```suggestion
                                "a NUMERIC column %s exceeds the maximum supported precision %d. Query failed.",
```